grc/core/generator/FlowGraphProxy.py

```diff
@@ -160,7 +160,7 @@ def links(self):
         Returns:
             a list of GNU Radio modules
         """
-        return [block.cpp_templates.render('link') for block in self.iter_enabled_blocks()]
+        return [block.cpp_templates.render('link') for block in self.iter_enabled_blocks() if not (block.is_virtual_sink() or block.is_virtual_source())]
```

Comment:
What problem does this solve exactly?

I'm asking because if this change (and the other virtual block-related changes you've made in this PR) is necessary for the virtual blocks, then I'm wondering if it might also be necessary for other blocks such as the embedded Python blocks.

Reply:
Without this modification, generating C++ with Virtual blocks produces the following error:
`Generate Error: 'VirtualSink' object has no attribute 'cpp_templates'`